code/frontend/src/components/UserProfile.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+function UserProfile() {
+  return (
+    <div>
+      <head>
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>User Profile</title>
+        <link rel="stylesheet" href="../styles/UserProfile.css" type="text/css" media="all" />
+      </head>
+      <body>
+        <header>
+          {/* Home Button */}
+          <div className="home-container">
+            <button id="home">🔄 Home</button>
+          </div>
+
+          {/* User Profile Container */}
+          <div className="profile-container">
+            <div className="profile-pic">
+              <div className="profile-pic">
+                <img
+                  src="https://www.shareicon.net/data/256x256/2016/09/15/829472_man_512x512.png"
+                  alt="Profile Image"
+                />{' '}
+                {/* Replace with your image */}
+              </div>
+            </div>
+            <span>Supriya Uppala</span>
+          </div>
+          <span></span> {/* Placeholder element */}
+        </header>
+
+        <section className="hobbies">
+          <h2>Hobbies:</h2>
+          <button>Singing</button>
+          <button>Dancing</button>
+          <button>Tennis</button>
+        </section>
+
+        <section className="video-history">
+          <h2>Video History</h2>
+          <div className="videos">
```

Comment:
Same issue with this. can you remove the hardcoding and make this code more dynamic? 
If I have 10 videos in an array...user would want to see all the 10 videos being displayed here.